### PR TITLE
LTR Faramir Prince of Ithilien

### PR DESCRIPTION
**Pull request by Simisays**
SVar:X:PlayerCountDefinedChosenPlayer$HasPropertyattackedYouTheirCurrentTurn isn't currently working

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+SVar:TrigChooseOpp:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | SubAbility$ DBDelayTrig
+SVar:DBDelayTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | ValidPlayer$ Player.IsTriggerRemembered | RememberObjects$ ChosenPlayer | Execute$ TrigBranch | TriggerDescription$ At the beginning of that player's next end step, you draw a card if they didn't attack you that turn. Otherwise, create three 1/1 white Human Soldier creature tokens.
+SVar:TrigBranch:DB$ Branch | BranchConditionSVar$ X | BranchConditionSVarCompare$ GE1 | TrueSubAbility$ DBToken | FalseSubAbility$ DBDraw | SubAbility$ DBCleanup
+SVar:DBDraw:DB$ Draw
+SVar:DBToken:DB$ Token | TokenScript$ w_1_1_human_soldier | TokenAmount$ 3 
+SVar:DBCleanup:DB$ Cleanup | ClearChosenPlayer$ True
+SVar:X:PlayerCountDefinedChosenPlayer$HasPropertyattackedYouTheirCurrentTurn
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt:12]: Try `PlayerCountDefinedTriggeredPlayer`

File: forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Name:Faramir, Prince of Ithilien
+ManaCost:2 W U
+Types:Legendary Creature Human Noble
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigChooseOpp | TriggerDescription$ At the beginning of your end step, choose an opponent. At the beginning of that player's next end step, you draw a card if they didn't attack you that turn. Otherwise, create three 1/1 white Human Soldier creature tokens.
+SVar:TrigChooseOpp:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | SubAbility$ DBDelayTrig
+SVar:DBDelayTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | ValidPlayer$ Player.IsTriggerRemembered | RememberObjects$ ChosenPlayer | Execute$ TrigBranch | TriggerDescription$ At the beginning of that player's next end step, you draw a card if they didn't attack you that turn. Otherwise, create three 1/1 white Human Soldier creature tokens.
+SVar:TrigBranch:DB$ Branch | BranchConditionSVar$ X | BranchConditionSVarCompare$ GE1 | TrueSubAbility$ DBToken | FalseSubAbility$ DBDraw | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt:8]: wrong cleanup placement

File: forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Types:Legendary Creature Human Noble
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigChooseOpp | TriggerDescription$ At the beginning of your end step, choose an opponent. At the beginning of that player's next end step, you draw a card if they didn't attack you that turn. Otherwise, create three 1/1 white Human Soldier creature tokens.
+SVar:TrigChooseOpp:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | SubAbility$ DBDelayTrig
+SVar:DBDelayTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | ValidPlayer$ Player.IsTriggerRemembered | RememberObjects$ ChosenPlayer | SubAbility$ DBCleanup | Execute$ TrigBranch | TriggerDescription$ At the beginning of that player's next end step, you draw a card if they didn't attack you that turn. Otherwise, create three 1/1 white Human Soldier creature tokens.
+SVar:TrigBranch:DB$ Branch | BranchConditionSVar$ X | BranchConditionSVarCompare$ GE1 | TrueSubAbility$ DBToken | FalseSubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw
+SVar:DBToken:DB$ Token | TokenScript$ w_1_1_human_soldier | TokenAmount$ 3 
+SVar:DBCleanup:DB$ Cleanup | ClearChosenPlayer$ True | ClearRemembered$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt:11]: there is nothing Remembered to clean up

File: forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt]: are you testing from this branch? because the support commit isn't in that yet

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt]: oh good point. It's not available yet from upstream on my end

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/faramir_prince_of_ithilien.txt]: Seems to work now thanks

**Review by tool4ever — APPROVED** [review]

**Pull request merged**